### PR TITLE
⏲️ Refactor complex timer with custom hook

### PR DESCRIPTION
**Pull request by kachick**
Current  timer has most complexity logics rooted on React/useRef/setInterval tech stacks.
So I have tried to refactor with custom hook, it hide the useRef and record/clear intervals. It makes easy `pausing`.

Honestly, I don't have strong confident, but looks working!⏲️ 

refs:
  * https://overreacted.io/making-setinterval-declarative-with-react-hooks/
  * https://usehooks-ts.com/react-hook/use-interval

### CODE REVIEW

File: src/common/storage.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ export const getStorageSoundEnabled = (): boolean => {
   return localStorage.getItem(KEY_SOUND_ENABLED) === "true";
 };
 
-export const setStorageSoundEnabled = (soundEnabled: boolean): void => {
-  localStorage.setItem(KEY_SOUND_ENABLED, JSON.stringify(soundEnabled));
+export const setStorageSoundEnabled = (isSoundEnabled: boolean): void => {
+  localStorage.setItem(KEY_SOUND_ENABLED, JSON.stringify(isSoundEnabled));
```
Comment (pankona) [inline src/common/storage.ts:10]: Good modify

**Review by pankona — APPROVED** [review]
I like it! 😋

**Comment by pankona** [comment]
Let's give it a try

**Comment by kachick** [comment]
Thank you!

File: src/common/storage.ts (hunk 1 above)
Comment (kachick) [reply src/common/storage.ts:10]: Honestly, I should split 😇

**Pull request merged**